### ISSUE TITLE
Unable to pull private images when using credentials helper

### ISSUE THREAD

**Issue by nkovacs** [docker/compose#3622]
I'm using the secretservice credentials helper with docker 1.11.2 and docker-compose 1.8.0-rc1.
If I try to run `docker-compose up`, it fails to pull private images from our gitlab registry. --verbose says:

```
docker.auth.auth.resolve_authconfig: No entry found
docker.api.image.pull: No auth config found
```

Pulling the image manually using docker pull works.

**Comment by nkovacs** [comment]
docker-compose version 1.8.0-rc1, build 9bf6bc6
docker-py version: 1.8.1
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013

**Comment by EugenMayer** [comment]
same here, same docker-compose version. Docker pull works, of course - cause is the credentials helper, i use

```
"credsStore": "osxkeychain"
```

docker-compose version 1.8.0-rc1
Docker version 1.11.2, build b9f10c9

**Comment by grosskur** [comment]
I think docker-py first needs to be updated to support the credential store API. See docker/docker-py/issues/1023. Then docker-compose can be updated to use the new docker-py.

**Comment by EugenMayer** [comment]
@grosskur docker is supporting credenial stores now, but compose is not. Is there any chance to move on?

**Comment by trevoro** [comment]
@shin- is there a proposal on fixing this? Would be happy to help.

**Comment by shin-** [comment]
@trevoro You can try out #3930 (or if you installed docker-compose via pip, just `pip install -U docker-py==1.10.2`) !

**Comment by trevoro** [comment]
👌

**Comment by jkinkead** [comment]
I have compose installed via Docker for Mac (Version 1.12.3 (13776)), and I'm seeing this issue.

```bash
$ docker-compose --version
docker-compose version 1.8.1, build 878cff1
```

Maybe the bundled `docker-py` isn't the right version?

**Comment by jkinkead** [comment]
Sorry, I take that  back: This bug is slightly different.

I have a `build` clause in my `docker-compose.yml` file, and the build step fails if the `FROM` image is in a private repository that isn't cached already.

`docker pull` works correctly.

**Comment by nkovacs** [comment]
That's probably the same bug. `docker pull` works, because that's docker, not docker-compose.

The fix should be in docker-compose 1.9, but I did not test it: https://github.com/docker/compose/commit/b64bd07f225dd8af9f2b8fd096dfd957022c3dda

**Comment by jkinkead** [comment]
Ahh. It looked like it was slated for `1.8.1`. Thanks!

**Comment by nkovacs** [comment]
Actually, it should be in 1.8.1 too: https://github.com/docker/compose/commit/e216a31f1e7d1c93844043802083197696dff74e

**Comment by jkinkead** [comment]
Hmm. I'll open a separate issue, then?

**Comment by shin-** [comment]
@jkinkead I think your issue is #4171, which will be fixed in 1.10.
